huaweicloud/services/ims/resource_huaweicloud_images_image_copy.go

```diff
@@ -0,0 +1,330 @@
+		_, err = images.Update(imsClient, d.Id(), updateOpts).Extract()
+		if err != nil {
+			return diag.Errorf("error updating image: %s", err)
+		}
+	}
+
+	// update tags
+	if d.HasChange("tags") {
+		tagErr := utils.UpdateResourceTags(imsClient, d, "images", d.Id())
+		if tagErr != nil {
+			return diag.Errorf("error updating tags of IMS image :%s, err:%s", d.Id(), tagErr)
+		}
+	}
+
+	return resourceImsImageCopyRead(ctx, d, meta)
+}
+
+func resourceImsImageCopyRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+
+	imsClient, err := getImsV2Client(d, cfg)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	img, err := GetCloudimage(imsClient, d.Id())
+	if err != nil {
+		return diag.Errorf("image %s not found: %s", d.Id(), err)
+	}
+	log.Printf("[DEBUG] Retrieved Image %s: %#v", d.Id(), img)
+
+	mErr := multierror.Append(
+		d.Set("region", region),
+		d.Set("name", img.Name),
+		d.Set("description", img.Description),
+		d.Set("kms_key_id", img.SystemCmkid),
+		d.Set("instance_id", getInstanceID(img.DataOrigin)),
+		d.Set("os_version", img.OsVersion),
+		d.Set("visibility", img.Visibility),
+		d.Set("data_origin", img.DataOrigin),
+		d.Set("disk_format", img.DiskFormat),
+		d.Set("image_size", img.ImageSize),
+		d.Set("checksum", img.Checksum),
+		d.Set("status", img.Status),
+		d.Set("enterprise_project_id", img.EnterpriseProjectID),
+	)
+
+	// fetch tags
+	if resourceTags, err := tags.Get(imsClient, "image", d.Id()).Extract(); err == nil {
+		tagMap := utils.TagsToMap(resourceTags.Tags)
+		mErr = multierror.Append(mErr, d.Set("tags", tagMap))
+	} else {
+		log.Printf("[WARN] Fetching tags of IMS images failed: %s", err)
+	}
+
+	return diag.FromErr(mErr.ErrorOrNil())
+}
+
+func resourceImsImageCopyDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+
+	imsClient, err := getImsV2Client(d, cfg)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	log.Printf("[DEBUG] Deleting Image %s", d.Id())
+	if err = images.Delete(imsClient, d.Id()).Err; err != nil {
+		return diag.Errorf("error deleting image copy: %s", err)
+	}
+
+	stateConf := &resource.StateChangeConf{
+		Pending:    []string{"ACTIVE"},
+		Target:     []string{"DELETED"},
+		Refresh:    waitForImageDelete(imsClient, d.Id()),
+		Timeout:    d.Timeout(schema.TimeoutDelete),
+		Delay:      5 * time.Second,
+		MinTimeout: 3 * time.Second,
+	}
+
+	_, err = stateConf.WaitForStateContext(ctx)
+	if err != nil {
+		return diag.Errorf("error waiting for delete image (%s) complete: %s", d.Id(), err)
+	}
+
+	d.SetId("")
+	return nil
+}
+
+func getImsV2Client(d *schema.ResourceData, cfg *config.Config) (*golangsdk.ServiceClient, error) {
+	imageRegion := common.GetRegion(d, cfg)
+	targetRegion := d.Get("target_region").(string)
+	if targetRegion != "" {
+		imageRegion = targetRegion
+	}
```

Comment:
```suggestion
	if v, ok := d.Get("target_region"); ok {
		imageRegion = v.(string)
	}
```

it's better to use d.GetOK

Reply:
done